app/src/main/java/org/thoughtcrime/securesms/database/Storage.kt

```diff
@@ -633,7 +633,11 @@ open class Storage(
                 // Notify the user
                 val threadID = getOrCreateThreadIdFor(Address.fromSerialized(groupId))
                 threadDb.setDate(threadID, formationTimestamp)
-                insertOutgoingInfoMessage(context, groupId, SignalServiceGroup.Type.CREATION, title, members.map { it.serialize() }, admins.map { it.serialize() }, threadID, formationTimestamp)
+
+                // Note: Commenting out this line prevents the timestamp of room creation being added to a new closed group,
+                // which in turn allows us to show the `groupNoMessages` control message text.
+                //insertOutgoingInfoMessage(context, groupId, SignalServiceGroup.Type.CREATION, title, members.map { it.serialize() }, admins.map { it.serialize() }, threadID, formationTimestamp)
```

Comment:
Isn't there a cost to commenting this out though? Don't we need that information?

Reply:
No, removing the line allows a new closed group to display `groupNoMessages` ("You have no messages from <b>{group_name}</b>. Send a message to start the conversation!") as the first entry for the newly joined member, then as soon as there _**are**_ any messages it is replaced with the timestamp of the last message, then it's timestamps all the way as per normal conversation.